framework/src/modules/chain/logic/transaction_pool.js

```diff
@@ -14,25 +14,46 @@
 
 'use strict';
 
-const async = require('async');
-// eslint-disable-next-line  prefer-const
-let jobsQueue = require('../helpers/jobs_queue.js');
-const transactionTypes = require('../helpers/transaction_types.js');
+const pool = require('@liskhq/lisk-transaction-pool');
+const { Status: TransactionStatus } = require('@liskhq/lisk-transactions');
 
 let modules;
-let library;
-let self;
 const {
 	EXPIRY_INTERVAL,
 	MAX_TRANSACTIONS_PER_BLOCK,
 	MAX_SHARED_TRANSACTIONS,
-	UNCONFIRMED_TRANSACTION_TIMEOUT,
 } = global.constants;
-const __private = {};
+
+const wrapAddTransactionResponseInCb = (
```

Comment:
Why callbacks instead of async/await or promises ?

Reply:
The wrapping functions by themselves return either synchronous response or return a promise. I am wrapping the return of those functions in callback because I wanted to keep the same interface as the old transaction pool.